src/storage/AddEdgesProcessor.cpp

```diff
@@ -13,27 +13,133 @@ namespace nebula {
 namespace storage {
 
 void AddEdgesProcessor::process(const cpp2::AddEdgesRequest& req) {
-    auto spaceId = req.get_space_id();
+    spaceId_ = req.get_space_id();
     auto version =
         std::numeric_limits<int64_t>::max() - time::WallClock::fastNowInMicroSec();
     // Switch version to big-endian, make sure the key is in ordered.
     version = folly::Endian::big(version);
 
     callingNum_ = req.parts.size();
+    auto iRet = schemaMan_->getEdgeIndexes(spaceId_);
+    if (iRet.ok()) {
+        for (auto& index : iRet.value()) {
+            indexes_.emplace_back(index);
+        }
+    }
     CHECK_NOTNULL(kvstore_);
-    std::for_each(req.parts.begin(), req.parts.end(), [&](auto& partEdges){
-        auto partId = partEdges.first;
-        std::vector<kvstore::KV> data;
-        std::for_each(partEdges.second.begin(), partEdges.second.end(), [&](auto& edge){
-            VLOG(4) << "PartitionID: " << partId << ", VertexID: " << edge.key.src
-                    << ", EdgeType: " << edge.key.edge_type << ", EdgeRanking: " << edge.key.ranking
-                    << ", VertexID: " << edge.key.dst << ", EdgeVersion: " << version;
-            auto key = NebulaKeyUtils::edgeKey(partId, edge.key.src, edge.key.edge_type,
-                                               edge.key.ranking, edge.key.dst, version);
-            data.emplace_back(std::move(key), std::move(edge.get_props()));
+    if (indexes_.empty()) {
+        std::for_each(req.parts.begin(), req.parts.end(), [&](auto& partEdges){
+            auto partId = partEdges.first;
+            std::vector<kvstore::KV> data;
+            std::for_each(partEdges.second.begin(), partEdges.second.end(), [&](auto& edge){
+                VLOG(4) << "PartitionID: " << partId << ", VertexID: " << edge.key.src
+                        << ", EdgeType: " << edge.key.edge_type << ", EdgeRanking: "
+                        << edge.key.ranking << ", VertexID: "
+                        << edge.key.dst << ", EdgeVersion: " << version;
+                auto key = NebulaKeyUtils::edgeKey(partId, edge.key.src, edge.key.edge_type,
+                                                   edge.key.ranking, edge.key.dst, version);
+                data.emplace_back(std::move(key), std::move(edge.get_props()));
+            });
+            doPut(spaceId_, partId, std::move(data));
         });
-        doPut(spaceId, partId, std::move(data));
+    } else {
+        std::for_each(req.parts.begin(), req.parts.end(), [&](auto& partEdges){
+            auto partId = partEdges.first;
+            const auto &edges = partEdges.second;
+            auto atomic = [&]() -> std::string {
```

Comment:
Maybe you should capture by value here, for example partId could be released.

Reply:
Sorry, I don't quite follow you. your means [&] ?